app/components/date-time-picker.js

```diff
@@ -38,29 +45,29 @@ export default class DateTimePicker extends Component{
   onChangeDate(isoDate, date) {
     let wasDateInputCleared = !date;
     if (!wasDateInputCleared) {
-      if(!this.date) {
-        this.date = new Date();
-      }
-      this.date.setDate(date.getDate());
-      this.date.setMonth(date.getMonth());
-      this.date.setFullYear(date.getFullYear());
+      let newDate = new Date(this.date.getTime());
+
+      newDate.setDate(date.getDate());
+      newDate.setMonth(date.getMonth());
+      newDate.setFullYear(date.getFullYear());
+
+      this.date = newDate;
       this.args.onChange(this.date);
     }
   }
 
   @action
   onChangeTime(type, event) {
     const value = event.target.value;
-    if(!this.date) {
-      this.date = new Date();
-    }
+    let newDate = new Date(this.date.getTime());
+
     if(type === 'hours') {
-      if(value < 0 || value > 24) return;
-      this.date.setHours(value);
+      newDate.setHours(value);
```

Comment:
This uses (abuses?) the fact that getHours will change the date when the value goes "out of bounds". 

- 24 -> date + 1, hour to 0
- -1 -> date - 1, hour to 23

It works and felt weird to implement manually, but it's not very obvious when reading the code that it happens.